src/msw/window.cpp

```diff
@@ -4809,6 +4831,27 @@ bool wxSystemParametersInfo(UINT uiAction, UINT uiParam, PVOID pvParam, UINT fWi
     return ::SystemParametersInfo(uiAction, uiParam, pvParam, fWinIni) == TRUE;
 }
 
+wxSize wxWindowMSW::GetDPI() const
+{
+    wxSize dpi = MSWTryGetWindowDPI(GetHwnd());
+
+    if ( !dpi.x || !dpi.y )
```

Comment:
Is it really possible to fail to get per-window DPI for this window itself but succeed with getting it for its TLW parent? I don't see when could this happen. If it really can, could you please add a comment here explaining it?

Reply:
I added it here: https://github.com/wxWidgets/wxWidgets/pull/334/commits/5066be542911e78bc7fd4d74b868f2c06ce154e7

I'll add this comment to the code.

Reply:
I.e. before actually creating the `HWND`? In this case we could just do it when `m_hWnd == 0` (and skip even trying to get per-window DPI in this case), couldn't we?

Reply:
Yes, it happens e.g. on the spin page in the widgets sample and `m_hwnd` is indeed NULL. So in that case tlw can be used directly.